core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -121,19 +121,16 @@ class RangePolicy : public Impl::PolicyTraits<Properties...> {
   // doesn't match.
   void operator()(const int&) const {}
 
-  RangePolicy(const RangePolicy&) = default;
-  RangePolicy(RangePolicy&&)      = default;
-
   template <class... OtherProperties>
-  RangePolicy(const RangePolicy<OtherProperties...> p) {
+  RangePolicy(const RangePolicy<OtherProperties...>& p) {
     m_space            = p.m_space;
     m_begin            = p.m_begin;
     m_end              = p.m_end;
     m_granularity      = p.m_granularity;
     m_granularity_mask = p.m_granularity_mask;
   }
 
-  inline RangePolicy() : m_space(), m_begin(0), m_end(0) {}
+  RangePolicy() noexcept : m_space(), m_begin(0), m_end(0) {}
```

Comment:
Actually this is interesting.  Why do we even provide a default constructor for `RangePolicy`?
As far as I can tell it is only used at these three places
https://github.com/kokkos/kokkos/blob/e31cb2fe241c2099fa84902fd8524a313ed882ad/core/src/Kokkos_Parallel.hpp#L196
https://github.com/kokkos/kokkos/blob/e31cb2fe241c2099fa84902fd8524a313ed882ad/core/src/Kokkos_Parallel.hpp#L440
https://github.com/kokkos/kokkos/blob/e31cb2fe241c2099fa84902fd8524a313ed882ad/core/src/Kokkos_Parallel.hpp#L517

Reply:
I plan on dropping that change because updating the exception specification on Cuda and HIP execution space is beyond the scope of this PR